### ISSUE TITLE
Empty aggregate report shows in non-multimodule projects

### ISSUE THREAD

**Issue by timp** [jacoco/jacoco#409]
### Steps to reproduce
JaCoCo version: 0.7.7-SNAPSHOT
Operating system: Window-7
Tool integration: Maven

### Expected behaviour
Only defined reports produced.

### Actual behaviour
Defined reports and an empty aggregate report are produced.

This is a recent change. 
See http://timp.github.io/juice/project-reports.html

I attempted to skip the report-aggregate goal but to no effect.
Report aggregate should not happen in a single level project, or at least only when more than one exec file is generated.

**Comment by marchof** [comment]
I just double checked: skip aktually works:

       <executions>
          <execution>
            <id>report-aggregate</id>
            <phase>verify</phase>
            <goals>
              <goal>report-aggregate</goal>
            </goals>
            <configuration>
              <skip>true</skip>
            </configuration>
          </execution>
        </executions>


If you don't want a aggregate report for your project then please don't configure one. The goal is designed to be configured for specific reporting reporting modules only. Due to the design of maven if it is not possible to e.g. simply enable it in the parent pom. We had a lengthy design discussion about this: https://github.com/jacoco/jacoco/wiki/MavenMultiModule

**Comment by marchof** [comment]
@timp Nice example project!

**Comment by timp** [comment]
@marchof thanks!

Your comments made me doubt myself, so I pulled the latest and rebuilt my snapshot. The build is not clean on java8/Windows7 but I still have the same result.

Hoping not to give offence but Skipping, as I have done it:
https://github.com/timp/juice/blob/master/pom.xml
does not work see 
http://timp.github.io/juice/project-reports.html

The reason why I added the skip is that without it the new empty aggregate report occurred when I used the snapshot. 
Now I am back home so have build, again, under openjdk8/ubuntu16.
Again jacoco does not build cleanly but after editing org.jacoco.core.test.pom.xml, skipping tests and killing hung builds I did get a new maven snapshot. (the magic to get a clean build might go well in the README.md)

The result is the same.

**Comment by Godin** [comment]
@timp about build: I guess that you're facing #300

**Comment by Godin** [comment]
@timp about `skip` parameter - I just checked your project:
```
$ git clone https://github.com/timp/juice.git
$ cd juice
$ mvn verify
...
23:32:26 [INFO] --- jacoco-maven-plugin:0.7.7-SNAPSHOT:report-aggregate (aggregate) @ juice ---
23:32:26 [INFO] Skipping JaCoCo execution because property jacoco.skip is set.
...
$ ls target/site 
jacoco-it  jacoco-unit
```
What am I doing wrong?

Also one of the points of @marchof - you don't need to declare `report-aggregate` if it anyway will be skipped, it is not bound to any phase by default so won't be executed without declaration ;)

**Comment by Godin** [comment]
@timp also instead of rebuilding snapshot you can grab it from snapshot repository, which always contains latest build of master - see http://eclemma.org/jacoco/trunk/doc/repo.html

**Comment by timp** [comment]
@godin I am building with 

    mvn clean install site -Dpublish

(Leave the -Dpublish off)
I have removed the execution for the aggregate-report (which contains the skip) locally but the result is unchanged: the empty aggregate report is still produced.

**Comment by Godin** [comment]
@timp okay, got it - Maven has a separate configuration/execution and so on for site reporting mojos: https://maven.apache.org/ref/3.3.9/maven-model/maven.html#class_reporting

**Comment by timp** [comment]
If this report was not empty but the aggregation of the UT and IT reports configured then this could be a feature not a bug.

**Comment by marchof** [comment]
@Godin So something we need to fix for the new goal?

@timp Please help us to fix one issue at a time -- let's focus on the report-aggregate goal here.

**Comment by timp** [comment]
Explicitly excluding the aggregate-report in a reportSet within the reporting section _does_ exclude the empty aggregate report
```
        <reportSets>
          <reportSet>
            <reports>
              <report>report</report>
              <report>report-integration</report>
            </reports>
          </reportSet>
        </reportSets>
```
The combined report, created with using the merge goal and a separate report execution, is not included in the reports and needs to be separately included from site.xml.
See http://timp.github.io/juice/project-reports.html

**Comment by marchof** [comment]
@timp So, we can close this?

**Comment by timp** [comment]
https://github.com/jacoco/jacoco/pull/418 ensures that the aggregate report is not produced if there are no data files to report upon, so removing the need to explicitly exclude it in a reportSet configuration. 

It may also fix an intermittent issue with file locking that was preventing a merge in the post-integration-test phase.

**Comment by Godin** [comment]
won't fix, see discussion in #418
